ROOTFS/usr/bin/waggle_network_watchdog.py

```diff
@@ -240,24 +245,26 @@ def hard_reboot_action():
             reboot_os()
 
     # Recovery actions table [time (s), recovery function]
-    # restart networking stack after 15, 20 and 25 of no beehive connectivity
-    # reboot after 30 mins of no beehive connectivity shutdown after 1 hour
-    # of no connection to beehive after 3 30 minute reboots
-    #
     # NOTE We sort in increasing order of threshold so that our linear
     # search finds the "earliest" available action
     recovery_actions = []
 
-    for time in nwwd_config.network_resets:
-        recovery_actions.append([time, reset_network_action])
+    # add in the soft and hard reboot actions
+    recovery_actions.append([nwwd_config.soft_reset_start, soft_reboot_action])
+    recovery_actions.append([nwwd_config.hard_reset_start, hard_reboot_action])
 
-    for time in nwwd_config.soft_resets:
-        recovery_actions.append([time, soft_reboot_action])
+    # we compute the number of network reset entries between the network restart
+    #  start time and the last possible action start time
+    last_action = nwwd_config.hard_reset_start
```

Comment:
Can we use something like:
```python3
last_action = min(nwwd_config.soft_reset_start, nwwd_config.hard_reset_start)
```

I think this will make the intent more clear if we revisit again far in the future.